dagql/server.go

```diff
@@ -592,6 +592,26 @@ func NewInstanceForCurrentID[P, T Typed](
 	}, nil
 }
 
+// InstanceWithNewSelector creates a new Instance that's set to the ID calculated
+// using new selector.
+//
+// This is useful in scenarios where we want to inject something into the selector
+// automatically without exposing it to the user. e.g. namespacing of the selector
+func InstanceWithNewSelector[P, T Typed](
```

Comment:
cc @sipsma - this is the function I added on top of `NewInstanceForCurrentID`.  It seems to have worked, but also I am not 100% sure if there are any side-effects of creating a new id by changing the selector like this.

Reply:
This comment is related: https://github.com/dagger/dagger/pull/8724#discussion_r1844746862

We might need something sort of different than this if the impurity is indeed wreacking havoc on the dagql caching